servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -5268,7 +5268,7 @@ void RendererSceneRenderRD::init() {
 			actions.renames["WORLD_POSITION"] = "world.xyz";
 			actions.renames["OBJECT_POSITION"] = "params.position";
 			actions.renames["UVW"] = "uvw";
-			actions.renames["EXTENTS"] = "params.extents";
```

Comment:
You have changed the name of the parameter, but you didn't change the name in the actual shader ([here](https://github.com/godotengine/godot/blob/eb98fd94421b124a5848d7ee6c7ead4337222c6c/servers/rendering/renderer_rd/shaders/volumetric_fog.glsl#L40)). So this isn't going to compile. Further, above you are still passing the extents to the shader as you convert size into an extent in the line just above. You will also need to convert the CPU-side uniform to size if you are going to change the shader parameter ([here](https://github.com/godotengine/godot/blob/eb98fd94421b124a5848d7ee6c7ead4337222c6c/servers/rendering/renderer_rd/renderer_scene_render_rd.h#L762))

Reply:
Sorry, yeah - I had problems with the shader code, partly because I don't fully understand how it's called yet (and I also don't have error detection/squiggles for GLSL currently)

Reply:
Would you mind pointing me towards where the values are passed into the relevant shaders please? I just can't seem to find it, so can't currently fix the input problem (I'm making the shaders use `extents` internally again).
By the way, should the visual shader expose `extents` (actually in the shader code) or `size` to the user?

Reply:
The visual shader should use the same variable names as the text shader. 

The CPU side variable is defined here: https://github.com/godotengine/godot/blob/eb98fd94421b124a5848d7ee6c7ead4337222c6c/servers/rendering/renderer_rd/renderer_scene_render_rd.h#L762

It is set here:
https://github.com/godotengine/godot/blob/eb98fd94421b124a5848d7ee6c7ead4337222c6c/servers/rendering/renderer_rd/renderer_scene_render_rd.cpp#L3964-L3966

And in the shader it is defined here:
https://github.com/godotengine/godot/blob/eb98fd94421b124a5848d7ee6c7ead4337222c6c/servers/rendering/renderer_rd/shaders/volumetric_fog.glsl#L40